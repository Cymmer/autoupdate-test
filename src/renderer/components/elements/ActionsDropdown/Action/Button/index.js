import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from '../styles.module.scss';

import Icon from '../../../Icon';
import Tooltip from '../../../Tooltip';
import tooltipPlacement from '../../../Tooltip/constants/tooltipPlacement';

const ActionButton = ({
  action: { id, action, icon, text, tooltip, disabled },
}) =>
  tooltip ? (
    <Tooltip content={tooltip} placement={tooltipPlacement.LEFT}>
      <button
        className={cn(styles.Action, {
          [styles.Action___disabled]: disabled,
          [styles.Action___delete]: icon === 'delete' || icon === 'close',
          [styles.Action___lock]: icon === 'lock',
        })}
        data-test="button"
        disabled={disabled}
        id={id}
        type="button"
        onClick={action}
      >
        <Icon className={styles.Action_icon} icon={icon} />
        {text}
      </button>
    </Tooltip>
  ) : (
    <button
      className={cn(styles.Action, {
        [styles.Action___disabled]: disabled,
        [styles.Action___delete]: icon === 'delete' || icon === 'close',
        [styles.Action___lock]: icon === 'lock',
      })}
      data-test="button"
      disabled={disabled}
      id={id}
      type="button"
      onClick={action}
    >
      <Icon className={styles.Action_icon} icon={icon} />
      {text}
    </button>
  );

ActionButton.propTypes = {
  action: PropTypes.shape({
    action: PropTypes.func,
    icon: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    disabled: PropTypes.bool,
    tooltip: PropTypes.string,
    id: PropTypes.string,
  }).isRequired,
};

export default ActionButton;
