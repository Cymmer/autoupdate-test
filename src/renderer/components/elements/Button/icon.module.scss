@import '../../../scss/variables';
@import '../../../scss/mixins';
@import '../../../scss/themes';

@mixin solid($size) {
  @include ease(0.3s);
  cursor: pointer;
  background-color: transparent;
  border-radius: 100px;
  border: none;
  text-transform: capitalize;
  outline: none;
  padding: 4px;
  height: max-content;
  text-decoration: none;
  @include flex-center;

  @include theme {
    color: t('iconButtonColor');

    &:hover {
      background-color: t('iconButtonBackgroundColor--hover');
      color: t('iconButtonColor--hover');
    }

    &:focus,
    &:active {
      background-color: t('iconButtonBackgroundColor--active');
    }
  }

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;

    &:hover,
    &:focus {
      opacity: 0.5 !important;
      background-color: transparent !important;
      color: $neutral-400 !important;
    }
  }

  .IconButton_icon {
    @if $size == xs {
      font-size: 1rem;
    } @else if $size == sm {
      font-size: 1.25rem;
    } @else if $size == md {
      font-size: 1.5rem;
    } @else if $size == lg {
      font-size: 2rem;
    }
  }
}

@mixin outline($size) {
  @include ease(0.3s);
  cursor: pointer;
  background-color: transparent;
  border-radius: 100px;
  text-transform: capitalize;
  outline: none;
  padding: 4px;
  height: max-content;
  text-decoration: none;
  @include flex-center;

  &:disabled {
    opacity: 0.5;
    cursor: default;

    &:hover,
    &:focus {
      background-color: transparent !important;
    }
  }

  .IconButton_icon {
    @if $size == xs {
      font-size: 1rem;
    } @else if $size == sm {
      font-size: 1.25rem;
    } @else if $size == md {
      font-size: 1.5rem;
    } @else if $size == lg {
      font-size: 2rem;
    }
  }
}

@mixin buttonColor($color) {
  color: $color;
  border: 1px solid $color;

  &:hover {
    background-color: transparentize($color, 0.9);
  }

  &:focus,
  &:active {
    background-color: transparentize($color, 0.75);
  }
}

.IconButton {
  &___solid {
    &___xs {
      @include solid(xs);
    }

    &___sm {
      @include solid(sm);
    }

    &___md {
      @include solid(md);
    }

    &___lg {
      @include solid(lg);
    }
  }

  &___outline {
    &___xs {
      @include outline(xs);
    }

    &___sm {
      @include outline(sm);
    }

    &___md {
      @include outline(md);
    }

    &___lg {
      @include outline(lg);
    }
  }

  &___disabled {
    opacity: 0.5;
    pointer-events: none;
  }

  &___blue {
    @include buttonColor($blue-300);
  }

  &___green {
    @include buttonColor($green-300);
  }

  &___red {
    @include buttonColor($red-300);
  }

  &_lockedIcon {
    position: absolute;
    bottom: 4px;
    right: 0px;
    font-size: 12px !important;
    color: $yellow-400 !important;

    @include theme {
      text-shadow: 0 0 2px t('cardBackgroundColor'),
        -1px -1px t('cardBackgroundColor');
    }
  }

  &_wrapper {
    position: relative;
    display: inline-block;
  }
}
