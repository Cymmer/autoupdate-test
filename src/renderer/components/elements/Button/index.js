import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import GLOBALS from 'codechum-app-globals';
import { STUDENT_ROUTES } from 'screen-wrappers/Student/constants';
import styles from './styles.module.scss';

import { Icon, Tooltip, Spinner } from '..';
import { spinnerSizes } from '../constants';

import buttonTypes from './constants/buttonTypes';

const Button = ({
  innerRef,
  children,
  type,
  kind,
  className,
  wrapperClassName,
  lockedIconClassName,
  iconClassName,
  onClick,
  disabled,
  icon,

  id,
  tabIndex,
  dataTestId,
  isLocked,
  isLoading,
}) => {
  const button = (
    <button
      ref={(e) => {
        if (innerRef) {
          innerRef.current = e;
        }
      }}
      className={cn(className, styles[`Button___${type}`], {
        [styles.Button___withIcon]: icon !== null,
        [styles.Button___disabled]: isLoading,
      })}
      data-test={dataTestId}
      disabled={disabled || isLocked || isLoading}
      id={id}
      tabIndex={tabIndex}
      type={kind}
      onClick={!isLocked ? onClick : () => {}}
    >
      {isLoading ? (
        <Spinner
          colorName={GLOBALS.COLOR_NAMES.WHITE}
          data-test="loadingSpinner"
          size={spinnerSizes.SM}
        />
      ) : (
        <>
          {icon && (
            <Icon
              className={cn(styles.Button___withIcon_icon, iconClassName)}
              icon={icon}
            />
          )}
          {children}
        </>
      )}
    </button>
  );

  return !isLocked ? (
    button
  ) : (
    <span
      className={cn(wrapperClassName, styles.Button_wrapper)}
      data-test={dataTestId}
    >
      {button}
      <Tooltip content={GLOBALS.MESSAGES.PREMIUM_FEATURE_LOCK_MESSAGE}>
        <a href={STUDENT_ROUTES.SHOP} rel="noreferrer" target="_blank">
          <Icon
            className={cn(styles.Button_lockedIcon, lockedIconClassName)}
            icon="lock"
          />
        </a>
      </Tooltip>
    </span>
  );
};

Button.defaultProps = {
  kind: GLOBALS.BUTTON_KINDS.BUTTON,
  type: buttonTypes.PRIMARY.BLUE,
  className: null,
  lockedIconClassName: null,
  wrapperClassName: null,
  disabled: false,
  id: null,
  icon: null,
  tabIndex: 0,
  onClick: () => {},
  isLocked: false,
  dataTestId: null,
  isLoading: false,
};

Button.propTypes = {
  kind: PropTypes.oneOf([
    GLOBALS.BUTTON_KINDS.BUTTON,
    GLOBALS.BUTTON_KINDS.SUBMIT,
    GLOBALS.BUTTON_KINDS.RESET,
  ]),
  type: PropTypes.oneOf([
    buttonTypes.PRIMARY.BLUE,
    buttonTypes.PRIMARY.RED,
    buttonTypes.PRIMARY.GREEN,
    buttonTypes.PRIMARY.YELLOW,
    buttonTypes.PRIMARY.PURPLE,
    buttonTypes.SECONDARY.BLUE,
    buttonTypes.SECONDARY.GREEN,
    buttonTypes.SECONDARY.RED,
    buttonTypes.SECONDARY.YELLOW,
    buttonTypes.SECONDARY.PURPLE,
    buttonTypes.SECONDARY.WHITE,
    buttonTypes.TEXT.BLUE,
    buttonTypes.TEXT.RED,
    buttonTypes.TEXT.GREEN,
    buttonTypes.TEXT.YELLOW,
    buttonTypes.TEXT.PURPLE,
    buttonTypes.TEXT.GRAY,
    buttonTypes.TERTIARY,
  ]),
  innerRef: PropTypes.any,
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  id: PropTypes.string,
  icon: PropTypes.string,
  tabIndex: PropTypes.number,
  dataTestId: PropTypes.string,
  isLocked: PropTypes.bool,
  iconClassName: PropTypes.string,

  // these can only have values if `isLocked` is true
  wrapperClassName: PropTypes.string,
  lockedIconClassName: PropTypes.string,
  isLoading: PropTypes.bool,
};

export default Button;
