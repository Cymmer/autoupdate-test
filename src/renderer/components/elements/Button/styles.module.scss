@import '../../../scss/variables';
@import '../../../scss/mixins';
@import '../../../scss/themes';

@mixin text($color) {
  @include base {
    color: $color;
    background: none;
    padding: 0;
    border: none;

    &:hover:not([disabled]) {
      color: darken($color, 5%);
    }

    &:focus:not([disabled]),
    &:active:not([disabled]) {
      color: darken($color, 15%);
    }

    &:disabled {
      @include theme {
        color: t('textColor-200');
      }
      opacity: 0.5;
      cursor: default;
    }
  }
}

.Button {
  &___primary {
    &___blue {
      @include primary($blue-300, $blue-400, $blue-500, $neutral-0);
    }

    &___red {
      @include primary($red-300, $red-400, $red-500, $neutral-0);
    }

    &___green {
      @include primary($green-300, $green-400, $green-500, $neutral-0);
    }

    &___yellow {
      @include primary($yellow-400, $yellow-400, $yellow-500, $neutral-0);
    }

    &___purple {
      @include primary($purple-300, $purple-300, $purple-300, $neutral-0);
    }
  }

  &___secondary {
    &___blue {
      @include secondary($blue-300);
    }

    &___red {
      @include secondary($red-300);
    }

    &___green {
      @include secondary($green-300);
    }

    &___yellow {
      @include secondary($yellow-400);
    }

    &___purple {
      @include secondary($purple-300);
    }

    &___white {
      @include secondary($neutral-0);
    }
  }

  &___tertiary {
    @include base {
      border-radius: 100px;
      padding: 10px 24px;
      border: none;

      @include theme {
        color: t('buttonTertiaryColor');
        background-color: t('buttonTertiaryBackgroundColor');

        &:hover:not([disabled]) {
          background-color: t('buttonTertiaryBackgroundColor--hover');
        }

        &:focus:not([disabled]),
        &:active:not([disabled]) {
          background-color: t('buttonTertiaryBackgroundColor--active');
        }

        &:disabled {
          opacity: 0.5;
          cursor: default;
        }
      }
    }
  }

  &___link {
    display: block;
  }

  &___text {
    &___blue {
      @include text($blue-300);
    }

    &___red {
      @include text($red-300);
    }

    &___green {
      @include text($green-300);
    }

    &___yellow {
      @include text($yellow-400);
    }

    &___purple {
      @include text($purple-300);
    }

    &___gray {
      @include text($neutral-300);
    }
  }

  &___withIcon {
    display: flex;
    align-items: center;
    justify-content: center;

    &_icon {
      margin-right: 8px;
    }
  }

  &___disabled {
    pointer-events: none;
    opacity: 0.5;
  }

  &_lockedIcon {
    position: absolute;
    bottom: -4px;
    right: -4px;
    font-size: 24px !important;
    color: $yellow-400 !important;

    @include theme {
      text-shadow: 0 0 2px t('cardBackgroundColor'),
        -1px -1px t('cardBackgroundColor');
    }
  }

  &_wrapper {
    position: relative;
    display: inline-block;
  }
}
