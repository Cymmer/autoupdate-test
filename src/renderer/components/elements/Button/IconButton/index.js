import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { Tooltip } from 'components/elements';
import GLOBALS from 'codechum-app-globals';

import { STUDENT_ROUTES } from 'screen-wrappers/Student/constants';

import Icon from '../../Icon';
import iconButtonTypes from '../constants/iconButtonTypes';

import styles from '../icon.module.scss';

const isTypeOutline = (type) => {
  switch (type) {
    case iconButtonTypes.OUTLINE.XS:
    case iconButtonTypes.OUTLINE.SM:
    case iconButtonTypes.OUTLINE.MD:
    case iconButtonTypes.OUTLINE.LG:
      return true;
    default:
      return false;
  }
};

const IconButton = ({
  icon,
  className,
  iconClassName,
  colorName,
  style,
  onClick,
  disabled,
  type,
  tabIndex,
  kind,
  id,
  isLocked,
}) => {
  const button = (
    <button
      className={cn(className, styles[`IconButton___${type}`], {
        [styles[`IconButton___${colorName}`]]: isTypeOutline(type),
      })}
      // eslint-disable-next-line
      data-test="button"
      disabled={isLocked || disabled}
      id={id}
      tabIndex={tabIndex}
      type={kind}
      onClick={disabled === false ? onClick || (() => {}) : null}
    >
      <Icon
        className={cn(styles.IconButton_icon, iconClassName)}
        icon={icon}
        style={style}
      />
    </button>
  );

  return !isLocked ? (
    button
  ) : (
    <span className={styles.IconButton_wrapper}>
      {button}
      <Tooltip content={GLOBALS.MESSAGES.PREMIUM_FEATURE_LOCK_MESSAGE}>
        <a href={STUDENT_ROUTES.SHOP} rel="noreferrer" target="_blank">
          <Icon className={styles.IconButton_lockedIcon} icon="lock" />
        </a>
      </Tooltip>
    </span>
  );
};

IconButton.defaultProps = {
  className: null,
  style: null,
  onClick: null,
  iconClassName: null,
  disabled: false,
  type: iconButtonTypes.SOLID.SM,
  kind: GLOBALS.BUTTON_KINDS.BUTTON,
  tabIndex: 0,
  id: null,
  colorName: GLOBALS.COLOR_NAMES.BLUE,
  isLocked: false,
};

IconButton.propTypes = {
  type: PropTypes.oneOf([
    iconButtonTypes.SOLID.LG,
    iconButtonTypes.SOLID.MD,
    iconButtonTypes.SOLID.SM,
    iconButtonTypes.SOLID.XS,
    iconButtonTypes.OUTLINE.LG,
    iconButtonTypes.OUTLINE.MD,
    iconButtonTypes.OUTLINE.SM,
    iconButtonTypes.OUTLINE.XS,
  ]),
  className: PropTypes.string,
  icon: PropTypes.string.isRequired,
  style: PropTypes.object,
  onClick: PropTypes.func,
  iconClassName: PropTypes.string,
  disabled: PropTypes.bool,
  tabIndex: PropTypes.number,
  kind: PropTypes.oneOf([
    GLOBALS.BUTTON_KINDS.BUTTON,
    GLOBALS.BUTTON_KINDS.SUBMIT,
    GLOBALS.BUTTON_KINDS.RESET,
  ]),
  id: PropTypes.string,
  colorName: PropTypes.oneOf([
    GLOBALS.COLOR_NAMES.BLUE,
    GLOBALS.COLOR_NAMES.RED,
    GLOBALS.COLOR_NAMES.GREEN,
  ]),
  isLocked: PropTypes.bool,
};

export default IconButton;
