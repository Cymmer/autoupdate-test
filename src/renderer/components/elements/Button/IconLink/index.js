import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import cn from 'classnames';
import GLOBALS from 'codechum-app-globals';

import { Tooltip } from 'components/elements';
import { STUDENT_ROUTES } from 'screen-wrappers/Student/constants';
import styles from '../icon.module.scss';

import Icon from '../../Icon';
import iconButtonTypes from '../constants/iconButtonTypes';

const isTypeOutline = (type) => {
  switch (type) {
    case iconButtonTypes.OUTLINE.XS:
    case iconButtonTypes.OUTLINE.SM:
    case iconButtonTypes.OUTLINE.MD:
    case iconButtonTypes.OUTLINE.LG:
      return true;
    default:
      return false;
  }
};

const handleClickIfLocked = (e) => {
  e.preventDefault();
};

const IconLink = ({
  icon,
  className,
  iconClassName,
  colorName,
  style,
  to,
  type,
  tabIndex,
  disabled,
  onClick,
  isLocked,
}) => {
  const link = (
    <Link
      className={cn(className, styles[`IconButton___${type}`], {
        [styles.IconButton___disabled]: isLocked || disabled,
        [styles[`IconButton___${colorName}`]]: isTypeOutline(type),
      })}
      data-test="link"
      tabIndex={tabIndex}
      to={to}
      onClick={isLocked ? handleClickIfLocked : onClick}
    >
      <Icon
        className={cn(styles.IconButton_icon, iconClassName)}
        icon={icon}
        style={style}
      />
    </Link>
  );
  return !isLocked ? (
    link
  ) : (
    <span className={styles.IconButton_wrapper}>
      {link}
      <Tooltip content={GLOBALS.MESSAGES.PREMIUM_FEATURE_LOCK_MESSAGE}>
        <a href={STUDENT_ROUTES.SHOP} rel="noreferrer" target="_blank">
          <Icon className={styles.IconButton_lockedIcon} icon="lock" />
        </a>
      </Tooltip>
    </span>
  );
};

IconLink.defaultProps = {
  className: null,
  style: null,
  to: null,
  iconClassName: null,
  type: iconButtonTypes.SOLID.SM,
  tabIndex: 0,
  disabled: false,
  colorName: GLOBALS.COLOR_NAMES.BLUE,
  onClick: null,
};

IconLink.propTypes = {
  type: PropTypes.oneOf([
    iconButtonTypes.SOLID.LG,
    iconButtonTypes.SOLID.MD,
    iconButtonTypes.SOLID.SM,
    iconButtonTypes.OUTLINE.LG,
    iconButtonTypes.OUTLINE.MD,
    iconButtonTypes.OUTLINE.SM,
  ]),
  className: PropTypes.string,
  icon: PropTypes.string.isRequired,
  style: PropTypes.object,
  to: PropTypes.string,
  iconClassName: PropTypes.string,
  tabIndex: PropTypes.number,
  disabled: PropTypes.bool,
  colorName: PropTypes.oneOf([
    GLOBALS.COLOR_NAMES.BLUE,
    GLOBALS.COLOR_NAMES.RED,
    GLOBALS.COLOR_NAMES.GREEN,
  ]),
  onClick: PropTypes.func,
  isLocked: PropTypes.bool,
};

export default IconLink;
