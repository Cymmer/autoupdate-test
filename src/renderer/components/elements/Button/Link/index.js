import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import cn from 'classnames';
import GLOBALS from 'codechum-app-globals';

import { Tooltip } from 'components/elements';
import { STUDENT_ROUTES } from 'screen-wrappers/Student/constants';

import styles from '../styles.module.scss';

import Icon from '../../Icon';
import buttonTypes from '../constants/buttonTypes';

const ButtonLink = ({
  children,
  type,
  className,
  to,
  icon,
  id,
  tabIndex,
  disabled,
  onClick,
  isLocked,
}) => {
  const allChildren = (
    <>
      {icon && (
        <Icon
          className={styles.Button___withIcon_icon}
          data-test="icon"
          icon={icon}
        />
      )}
      {children}
    </>
  );

  const link = to.startsWith('http') ? (
    <a
      className={cn(
        className,
        styles.Button___link,
        styles[`Button___${type}`],
        {
          [styles.Button___withIcon]: icon !== null,
          [styles.Button___disabled]: isLocked || disabled,
        }
      )}
      data-test="nativeAnchor"
      href={to}
      id={id}
      rel="noreferrer"
      tabIndex={tabIndex}
      target="_blank"
      onClick={onClick}
    >
      {allChildren}
    </a>
  ) : (
    <Link
      className={cn(
        className,
        styles.Button___link,
        styles[`Button___${type}`],
        {
          [styles.Button___withIcon]: icon !== null,
          [styles.Button___disabled]: isLocked || disabled,
        }
      )}
      data-test="reactRouterLink"
      id={id}
      tabIndex={tabIndex}
      to={to}
      onClick={onClick}
    >
      {allChildren}
    </Link>
  );

  return !isLocked ? (
    link
  ) : (
    <span className={styles.Button_wrapper}>
      {link}
      <Tooltip content={GLOBALS.MESSAGES.PREMIUM_FEATURE_LOCK_MESSAGE}>
        <a href={STUDENT_ROUTES.SHOP} rel="noreferrer" target="_blank">
          <Icon className={styles.Button_lockedIcon} icon="lock" />
        </a>
      </Tooltip>
    </span>
  );
};

ButtonLink.defaultProps = {
  type: buttonTypes.PRIMARY.BLUE,
  className: null,
  id: null,
  icon: null,
  tabIndex: 0,
  disabled: false,
  onClick: null,
  isLocked: false,
};

ButtonLink.propTypes = {
  type: PropTypes.oneOf([
    buttonTypes.PRIMARY.BLUE,
    buttonTypes.PRIMARY.RED,
    buttonTypes.PRIMARY.GREEN,
    buttonTypes.SECONDARY.BLUE,
    buttonTypes.SECONDARY.GREEN,
    buttonTypes.SECONDARY.RED,
    buttonTypes.TEXT.BLUE,
    buttonTypes.TEXT.RED,
    buttonTypes.TEXT.GREEN,
    buttonTypes.TERTIARY,
  ]),
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  to: PropTypes.string.isRequired,
  id: PropTypes.string,
  icon: PropTypes.string,
  tabIndex: PropTypes.number,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  isLocked: PropTypes.bool,
};

export default ButtonLink;
