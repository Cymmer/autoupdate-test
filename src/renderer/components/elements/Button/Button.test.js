import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './index';

describe('RENDERING - Button', () => {
  it('should render with default props', () => {
    const component = shallow(<Button onClick={() => {}}>Test</Button>);
    expect(component).toMatchSnapshot();
  });

  it('should render an icon', () => {
    const component = shallow(
      <Button icon="test" onClick={() => {}}>
        Test
      </Button>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render a spinner if loading', () => {
    const component = shallow(
      <Button isLoading onClick={() => {}}>
        Test
      </Button>
    );
    expect(component).toMatchSnapshot();
  });
});

describe('EVENTS - Button', () => {
  it('should be clickable if not disabled', () => {
    const clickFn = jest.fn();
    const component = mount(
      <Button disabled={false} icon="test" onClick={clickFn}>
        Test
      </Button>
    );

    component.simulate('click');

    expect(clickFn).toHaveBeenCalled();
  });

  it('should not be clickable if disabled', () => {
    const clickFn = jest.fn();
    const component = mount(
      <Button disabled icon="test" onClick={clickFn}>
        Test
      </Button>
    );

    component.simulate('click');

    expect(clickFn).not.toHaveBeenCalled();
  });

  it('should not be clickable if loading', () => {
    const clickFn = jest.fn();
    const component = mount(
      <Button isLoading icon="test" onClick={clickFn}>
        Test
      </Button>
    );

    component.simulate('click');

    expect(clickFn).not.toHaveBeenCalled();
  });
});
