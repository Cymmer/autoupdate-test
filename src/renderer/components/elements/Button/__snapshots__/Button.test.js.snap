// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Button should render a spinner if loading 1`] = `
<button
  className="Button___primary___blue Button___disabled"
  data-test={null}
  disabled={true}
  id={null}
  onClick={[Function]}
  tabIndex={0}
  type="button"
>
  <Spinner
    className={null}
    colorName="white"
    data-test="loadingSpinner"
    size="sm"
  />
</button>
`;

exports[`RENDERING - Button should render an icon 1`] = `
<button
  className="Button___primary___blue Button___withIcon"
  data-test={null}
  disabled={false}
  id={null}
  onClick={[Function]}
  tabIndex={0}
  type="button"
>
  <Icon
    className="Button___withIcon_icon"
    icon="test"
    id={null}
    isPulsating={false}
    style={null}
  />
  Test
</button>
`;

exports[`RENDERING - Button should render with default props 1`] = `
<button
  className="Button___primary___blue"
  data-test={null}
  disabled={false}
  id={null}
  onClick={[Function]}
  tabIndex={0}
  type="button"
>
  Test
</button>
`;
